card/mod/01_core/set/all/name.rb

```diff
@@ -228,60 +251,73 @@ def repair_key
 end
 
-
 event :cascade_name_changes, after: :store, on: :update, changed: :name do
-  #Rails.logger.info "------------------- #{name_was} CASCADE #{self.name} -------------------------------------"
+  # Rails.logger.info "------------------- #{name_was} CASCADE #{self.name} " \
+  #                   " -------------------------------------"
 
-  self.update_referencers = false if self.update_referencers == 'false' #handle strings from cgi
+  # handle strings from cgi
+  update_referencers = false if update_referencers == 'false'
   Card::Reference.update_on_rename self, name, self.update_referencers
 
-  deps = self.dependents
-  #warn "-------------------#{name_was}---- CASCADE #{self.name} -> deps: #{deps.map(&:name)*", "} -----------------------"
+  deps = dependents
+  # warn "-------------------#{name_was}---- CASCADE #{self.name} -> deps: " \
+  #      " #{deps.map(&:name)*', '} -----------------------"
 
-  @dependents = nil #reset
+  @dependents = nil # reset
 
   deps.each do |dep|
-    # here we specifically want NOT to invoke recursive cascades on these cards, have to go this low level to avoid callbacks.
+    # here we specifically want NOT to invoke recursive cascades on these
+    # cards, have to go this low level to avoid callbacks.
     Rails.logger.info "cascading name: #{dep.name}"
-    Card.expire dep.name #old name
+    Card.expire dep.name # old name
     newname = dep.cardname.replace_part name_was, name
-    Card.where( id: dep.id ).update_all name: newname.to_s, key: newname.key
+    Card.where(id: dep.id).update_all name: newname.to_s, key: newname.key
     Card::Reference.update_on_rename dep, newname, update_referencers
     Card.expire newname
   end
-  if update_referencers
-    Auth.as_bot do
-      [self.name_referencers(name_was)+(deps.map &:referencers)].flatten.uniq.each do |card|
-        # FIXME  using "name_referencers" instead of plain "referencers" for self because there are cases where trunk and tag
-        # have already been saved via association by this point and therefore referencers misses things
-        # eg.  X includes Y, and Y is renamed to X+Z.  When X+Z is saved, X is first updated as a trunk before X+Z gets to this point.
-        # so at this time X is still including Y, which does not exist.  therefore #referencers doesn't find it, but name_referencers(old_name) does.
-        # some even more complicated scenario probably breaks on the dependents, so this probably needs a more thoughtful refactor
-        # aligning the dependent saving with the name cascading
-
-        Rails.logger.debug "------------------ UPDATE REFERER #{card.name}  ------------------------"
-        unless card == self or card.structure
-          card = card.refresh
-          card.db_content = card.replace_references name_was, name
-          card.save!
-        end
+  execute_referencers_update(deps) if update_referencers
+end
+
+def execute_referencers_update dependents
```

Comment:
Assignment Branch Condition size for execute_referencers_update is too high. [19.13/15]